katran/lib/KatranLb.cpp

```diff
@@ -1769,8 +1816,10 @@ bool KatranLb::updateVipMap(
   return true;
 }
 
-bool KatranLb::updateRealsMap(const folly::IPAddress& real, uint32_t num) {
+bool KatranLb::updateRealsMap(const folly::IPAddress& real, uint32_t num, uint8_t flags) {
   auto real_addr = IpHelpers::parseAddrToBe(real);
+  flags &= ~V6DADDR; // to keep IPv4/IPv6 specific header
```

Comment:
updateReals are either called from increaseRefCount of modifyReals. both of em already taking care about this flag

Reply:
For sure about this in future developments. But is not necessary. I'll remove it.